src/Microsoft.IdentityModel.Tokens/SupportedAlgorithms.cs

```diff
@@ -170,11 +171,51 @@ internal static bool IsSupportedRsaAlgorithm(string algorithm)
                 case SecurityAlgorithms.RsaPKCS1:
                 case SecurityAlgorithms.RsaOaepKeyWrap:
                     return true;
+                case SecurityAlgorithms.RsaSsaPssSha256:
+                case SecurityAlgorithms.RsaSsaPssSha384:
+                case SecurityAlgorithms.RsaSsaPssSha512:
+                case SecurityAlgorithms.RsaSsaPssSha256Signature:
+                case SecurityAlgorithms.RsaSsaPssSha384Signature:
+                case SecurityAlgorithms.RsaSsaPssSha512Signature:
+                    return IsSupportedRsaPss(key);
             }
 
             return false;
         }
 
+        private static bool IsSupportedRsaPss(SecurityKey key)
+        {
+#if NET45 || NET451
+            // RSA-PSS is not available on .NET 4.5 and .NET 4.5.1
+            LogHelper.LogExceptionMessage(new PlatformNotSupportedException(LogMessages.IDX10692));
+            return false;
+#elif NET461 || NETSTANDARD2_0
+            // RSACryptoServiceProvider doesn't support RSA-PSS
+            if (key is RsaSecurityKey rsa && rsa.Rsa is RSACryptoServiceProvider)
+            {
+                LogHelper.LogExceptionMessage(new PlatformNotSupportedException(LogMessages.IDX10693));
+                return false;
+            }
+            else if (key is X509SecurityKey x509 && x509.PrivateKey is RSACryptoServiceProvider)
```

Comment:
Public and Private keys will return the same RSA type. We only need to check the Public key.

Reply:
addressed with daa3d717fd9d56f3d246d0f5c329005b88f7872b